File: 169. 多数元素.py
class Solution:
    def majorityElement(self, nums):
        star = 0
        end = len(nums)-1
        while star < end:
            if nums[star] != nums[end]:
                del nums[star]
                del nums[end]
            star += 1
            end -= 1
        print(nums)
        return nums[0]

class Solution1:
    def majorityElement(self, nums):
        nums.sort()         #先排序
        return nums[len(nums)//2]           #中间的地方一定就是
nums = [2,2,1,1,1,2,2]
print(Solution1().majorityElement(nums))

File: 169. 多数元素.java
class Solution {
    public int majorityElement(int[] nums) {
        //将所有不相等的元素对去掉，剩下的就是。
        //众数出现的次数比非众数多（能抵消掉）
        int count = 0;
        int candidate = null;
        for(int i=0; i<nums.length; i++){
            if(count==0) candidate = nums[i];       //更改候选
            count += (candidate==nums[i])?1:-1; 
        }
        return candidate;
    }
}

